src/Resolver/RedirectRouteResolver.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusGiftCardPlugin\Resolver;
+
+use function array_key_exists;
+use Exception;
+use const FILTER_SANITIZE_URL;
+use function filter_var;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+final class RedirectRouteResolver implements RedirectRouteResolverInterface
+{
+    /** @var UrlGeneratorInterface */
+    private $router;
+
+    public function __construct(UrlGeneratorInterface $router)
+    {
+        $this->router = $router;
+    }
+
+    public function getRouteToRedirectTo(Request $request, string $defaultRoute): string
+    {
+        if ($request->attributes->has('redirect')) {
+            $redirect = $request->attributes->get('redirect');
+
+            try {
+                if (array_key_exists(0, $redirect)) {
+                    return $this->router->generate($redirect[0]);
+                }
+                if (array_key_exists('route', $redirect) && array_key_exists('parameters', $redirect)) {
+                    return $this->router->generate($redirect['route'], $redirect['parameters']);
+                }
+            } catch (Exception $exception) {
```

Comment:
This exception is throw is the route doesn't exist, right? If yes, I don't think we should catch it. For now I think it's better to surface this error, so a developer can fix the error

Reply:
Yes, that's right. Ok, it makes sense. I thought it would be more permissive this way, but that just sounds useless and it can only raise weird behaviors